src/MPDevice.cpp

```diff
@@ -5922,7 +5951,13 @@ void MPDevice::loadFreeAddresses(AsyncJobs *jobs, const QByteArray &addressFrom,
 {
     qDebug() << "Loading free addresses from address:" << addressFrom.toHex();
 
-    jobs->append(new MPCommandJob(this, MPCmd::GET_30_FREE_SLOTS,
+    if (isBLE())
+    {
+        bleImpl->getFreeAddressProvider().loadFreeAddresses(jobs, addressFrom, cbProgress);
```

Comment:
are you discarding discardFirstAddr? this is important as you'll get twice the same address in your buffer if not.